front/src/pages/Category.tsx

```diff
@@ -502,30 +584,94 @@ const Category: FC = () => {
               </FiltersContainer>
               <ResourcesContainer>
                 <TitleResourcesContainer>
-                  <Title as="h2" fontWeight="bold">
-                  {translation('Recursos de', { name: state?.name })}
-                  </Title>
-                  <SearchBar
-                    data-testid="inputGroupSearch"
-                    label="searchResource"
-                    name="searchResource"
-                    placeholder={translation('Buscar recurso')}
-                    id="searchResource"
-                    icon="search"
-                  />
+                  {isSearch ? (
+                    <SearchContainer>
+                      <Title as="h2" fontWeight="bold">
+                        {t('Buscar recurso')}
+                      </Title>
+                      <InputSearchBar >
+                        <InputSearch
+                          type="text"
+                          data-testid="inputSearch"
+                          onChange={(e) => setSearchValue(e.target.value)}
+                        />
+                        <CancelSearchButton onClick={toggleSearch} data-testid="cancelSearchButton">
+                          X
+                        </CancelSearchButton>
+                      </InputSearchBar>
+                      {searchValue !== null && searchValue !== '' ? (
+                        <span style={{ marginTop: '20px', fontWeight: 'bold' }}>
+                          {t('Mostrando')} {selectedSortOrderValue.length} {t('resultados para')} &quot;
+                          {searchValue}&quot;
+                        </span>
+                      ) : null}
+                    </SearchContainer>
+                  ) : (
+                    <>
+                      <Title as="h2" fontWeight="bold">
+                        {t('Recursos de')} {state?.name}
+                      </Title>
+                      <SearchBar
+                        data-testid="inputGroupSearch"
+                        label="searchResource"
+                        name="searchResource"
+                        placeholder={t("Buscar recurso")}
+                        id="searchResource"
+                        icon="search"
+                        onClick={toggleSearch}
+                      />
+                    </>
+                  )}
+
                   <FilterButton
                     data-testid="filters-button"
                     onClick={handleFiltersOpen}
                   >
-                    {translation('Filtrar')}
+                    {t('Filtrar')}
                   </FilterButton>
                 </TitleResourcesContainer>
-                <VotesDateController
-                  sortOrder={sortOrder}
-                  handleSortOrder={handleSortOrder}
-                  handleSortByVotes={handleSortByVotes}
-                  handleSortByDates={handleSortByDates}
-                />
+                <VotesDateContainer>
```

Comment:
Great job so far! However, I noticed that the recent changes introducing 'VotesDateController' were reverted, and 'VotesDateContainer' is back in use. Please note that 'VotesDateController' contains all the previous logic and styles while staying as an isolated component now!